tests/gpflow/experimental/check_shapes/test_parser.py

```diff
@@ -745,19 +747,155 @@ def __str__(self) -> str:
 
 @pytest.mark.parametrize("data", _TEST_DATA, ids=str)
 def test_parse_argument_spec(data: TestData) -> None:
-    actual_specs = tuple(parse_argument_spec(s) for s in data.argument_spec_strs)
+    actual_specs = tuple(parse_argument_spec(s, TestContext()) for s in data.argument_spec_strs)
     assert data.expected_specs == actual_specs
 
 
 @pytest.mark.parametrize("data", _TEST_DATA, ids=str)
 def test_parse_and_rewrite_docstring(data: TestData) -> None:
-    rewritten_docstring = parse_and_rewrite_docstring(data.doc, data.expected_specs)
+    rewritten_docstring = parse_and_rewrite_docstring(data.doc, data.expected_specs, TestContext())
     assert data.expected_doc == rewritten_docstring
 
 
 @pytest.mark.parametrize("data", _TEST_DATA, ids=str)
 def test_parse_and_rewrite_docstring__disable(data: TestData) -> None:
     set_rewrite_docstrings(None)
 
-    rewritten_docstring = parse_and_rewrite_docstring(data.doc, data.expected_specs)
+    rewritten_docstring = parse_and_rewrite_docstring(data.doc, data.expected_specs, TestContext())
     assert data.doc == rewritten_docstring
+
+
+_CHECK_SHAPES_LINE = 892
+
+
+@pytest.mark.parametrize(
+    "spec,expected_message",
+    [
+        (
+            "a [batch..., x]",
+            f"""
+Unable to parse shape specification.
+  check_shapes called at: {__file__}:{_CHECK_SHAPES_LINE}
+    Argument number (0-indexed): 0
+      Line:     "a [batch..., x]"
+                    ^
+      Expected: integer (re=(?:[0-9])+)
```

Comment:
(a pity it can't infer the missing : but I guess it's true that it expects an integer now)

Reply:
> (a pity it can't infer the missing : but I guess it's true that it expects an integer now)

Yeah, it's limited what the poor LALR parsers can do.